### PR TITLE
Fix some clippy issues

### PR DESCRIPTION
**Pull request by alex-semenyuk**
Run and fix some issues from `cargo clippy`

### CODE REVIEW

**Comment by ytmimi** [comment]
Out of curiosity, which clippy lints were fixed by this PR?

File: src/cargo-fmt/main.rs [hunk 1]
```diff
@@ -198,12 +198,10 @@ fn convert_message_format_to_rustfmt_args(
             Ok(())
         }
         "human" => Ok(()),
-        _ => {
-            return Err(format!(
-                "invalid --message-format value: {}. Allowed values are: short|json|human",
-                message_format
-            ));
-        }
+        _ => Err(format!(
```
Comment (alex-semenyuk) [inline src/cargo-fmt/main.rs:201]: warning: unneeded `return` statement
```
  --> src/lib.rs:43:13
   |
43 | /             return Err(format!(
44 | |                 "invalid --message-format value: {}. Allowed values are: short|json|human",
45 | |                 message_format
46 | |             ));
   | |_______________^
   |
```
   = note: `#[warn(clippy::needless_return)]` on by default
   = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#needless_return

File: src/cargo-fmt/main.rs [hunk 2]
```diff
@@ -215,7 +213,7 @@ fn print_usage_to_stderr(reason: &str) {
         .expect("failed to write to stderr");
 }
 
-#[derive(Debug, Clone, Copy, PartialEq)]
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
```
Comment (alex-semenyuk) [inline src/cargo-fmt/main.rs:216]: warning: you are deriving `PartialEq` and can implement `Eq`
```
 --> src/lib.rs:1:30
  |
1 | #[derive(Debug, Clone, Copy, PartialEq)]
  |                              ^^^^^^^^^ help: consider deriving `Eq` as well: `PartialEq, Eq`
  |
```
  = note: `#[warn(clippy::derive_partial_eq_without_eq)]` on by default
  = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#derive_partial_eq_without_eq

File: src/cargo-fmt/test/mod.rs [hunk 3]
```diff
@@ -70,9 +70,9 @@ fn mandatory_separator() {
             .is_err()
     );
     assert!(
-        !Opts::command()
+        Opts::command()
```
Comment (alex-semenyuk) [inline src/cargo-fmt/test/mod.rs:73]: warning: this boolean expression can be simplified
  |
  = note: `#[warn(clippy::nonminimal_bool)]` on by default
  = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#nonminimal_bool

**Comment by ytmimi** [comment]
@alex-semenyuk thanks for taking the time to outline all that!

**Comment by ytmimi** [comment]
Can you rebase these changes on the current master? The branch needs to be up to date for me to merge these changes. Also, if possible can you squash this into a single commit.

**Comment by alex-semenyuk** [comment]
@ytmimi Sure, ready

**Review by ytmimi — APPROVED** [review]
Thanks for making these changes!

**Pull request merged**